sdk/core/src/lro.rs

```diff
@@ -0,0 +1,127 @@
+/// * `Retry-After`
+/// * `retry-after-ms`
+/// * `x-ms-retry-after-ms`
+///
+/// If no header is provided, the default retry time will be returned.
+pub fn get_retry_after(headers: &Headers) -> Duration {
+    [RETRY_AFTER, RETRY_AFTER_MS, X_MS_RETRY_AFTER_MS]
+        .iter()
+        .find_map(|header| {
+            headers
+                .get_str(header)
+                .ok()
+                .and_then(|v| v.parse::<u64>().ok())
+                .map(Duration::from_secs)
+        })
+        .unwrap_or(DEFAULT_RETRY_TIME)
+}
+
+pub mod location {
+    use crate::{
+        headers::{Headers, AZURE_ASYNCOPERATION, LOCATION, OPERATION_LOCATION},
+        lro::LroStatus,
+        Url,
+    };
+
+    #[derive(Debug)]
+    pub enum FinalState {
+        AzureAsyncOperation,
+        Location,
+        OperationLocation,
+    }
+
+    pub fn get_location(headers: &Headers, final_state: FinalState) -> crate::Result<Option<Url>> {
+        match final_state {
+            FinalState::AzureAsyncOperation => headers.get_optional_as(&AZURE_ASYNCOPERATION),
+            FinalState::Location => headers.get_optional_as(&LOCATION),
+            FinalState::OperationLocation => headers.get_optional_as(&OPERATION_LOCATION),
+        }
+    }
+
+    pub fn get_provisioning_state(body: &[u8]) -> Option<LroStatus> {
+        let body: serde_json::Value = serde_json::from_slice(body).ok()?;
+        let provisioning_state = body["status"].as_str()?;
+        Some(LroStatus::from(provisioning_state))
+    }
+}
+
+pub mod body_content {
+    use crate::{lro::LroStatus, StatusCode};
+    use serde::Serialize;
+
+    /// Extract the provisioning state based on the status code and response body
+    ///
+    /// Ref: <https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/core/azure-core/azure/core/polling/base_polling.py>
+    pub fn get_provisioning_state<S>(status_code: StatusCode, body: &S) -> crate::Result<LroStatus>
+    where
+        S: Serialize,
+    {
+        match status_code {
+            StatusCode::Accepted => Ok(LroStatus::InProgress),
+            StatusCode::Created => {
+                Ok(get_provisioning_state_from_body(body).unwrap_or(LroStatus::InProgress))
+            }
+            StatusCode::Ok => {
+                Ok(get_provisioning_state_from_body(body).unwrap_or(LroStatus::Succeeded))
+            }
+            StatusCode::NoContent => Ok(LroStatus::Succeeded),
+            _ => Err(crate::error::Error::from(
+                crate::error::ErrorKind::HttpResponse {
+                    status: status_code,
+                    error_code: Some("invalid status found in LRO response".to_owned()),
+                },
+            )),
+        }
+    }
+
+    fn get_provisioning_state_from_body<S>(body: &S) -> Option<LroStatus>
+    where
+        S: Serialize,
+    {
+        let body: serde_json::Value =
+            serde_json::from_str(&serde_json::to_string(body).ok()?).ok()?;
```

Comment:
What is this doing?

Reply:
The simplest way I could come up with to generically get the json path, regardless of the underlying response type was to parse it here, then reparse it by the caller.   

My thought is this could eventually be retooled to be a macro or a trait, but this was significantly simpler to get functional.